### PR TITLE
Do not set Name attributes if unsupported

### LINKED ISSUE #2655: Do not set names if the solver does not support it
MWE 

```julia

julia>model = direct_model(Ipopt.Optimizer())
julia> @variable(model, x[i=1:2] >= 0)
ERROR: MathOptInterface.UnsupportedAttribute{MathOptInterface.VariableName}: Attribute MathOptInterface.VariableName() is not supported by the model.
Stacktrace:
  [1] throw_set_error_fallback(model::Ipopt.Optimizer, attr::MathOptInterface.VariableName, index::MathOptInterface.VariableIndex, value::String; error_if_supported::MathOptInterface.SetAttributeNotAllowed{MathOptInterface.VariableName})
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/attributes.jl:438
  [2] throw_set_error_fallback(model::Ipopt.Optimizer, attr::MathOptInterface.VariableName, index::MathOptInterface.VariableIndex, value::String)
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/attributes.jl:435
  [3] set(::Ipopt.Optimizer, ::MathOptInterface.VariableName, ::MathOptInterface.VariableIndex, ::String)
    @ MathOptInterface ~/.julia/packages/MathOptInterface/YDdD3/src/attributes.jl:410
  [4] set(model::Model, attr::MathOptInterface.VariableName, v::VariableRef, value::String)
    @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/JuMP.jl:1151
  [5] set_name
    @ ~/.julia/packages/JuMP/Xrr7O/src/variables.jl:315 [inlined]
  [6] _moi_add_variable(backend::Ipopt.Optimizer, model::Model, v::ScalarVariable{Int64, Float64, Float64, Float64}, name::String)
    @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/variables.jl:971
  [7] add_variable(model::Model, v::ScalarVariable{Int64, Float64, Float64, Float64}, name::String)
    @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/variables.jl:963
  [8] (::var"#45#46")(i::Int64)
    @ Main ~/.julia/packages/JuMP/Xrr7O/src/Containers/macro.jl:206
  [9] jump-dev/Ipopt.jl#31
    @ ~/.julia/packages/JuMP/Xrr7O/src/Containers/container.jl:72 [inlined]
 [10] iterate
    @ ./generator.jl:47 [inlined]
 [11] collect(itr::Base.Generator{JuMP.Containers.VectorizedProductIterator{Tuple{Base.OneTo{Int64}}}, JuMP.Containers.var"#31#32"{var"#45#46"}})
    @ Base ./array.jl:678
 [12] map(f::Function, A::JuMP.Containers.VectorizedProductIterator{Tuple{Base.OneTo{Int64}}})
    @ Base ./abstractarray.jl:2323
 [13] container
    @ ~/.julia/packages/JuMP/Xrr7O/src/Containers/container.jl:72 [inlined]
 [14] container(f::Function, indices::JuMP.Containers.VectorizedProductIterator{Tuple{Base.OneTo{Int64}}})
    @ JuMP.Containers ~/.julia/packages/JuMP/Xrr7O/src/Containers/container.jl:66
 [15] macro expansion
    @ ~/.julia/packages/JuMP/Xrr7O/src/macros.jl:142 [inlined]
 [16] top-level scope
    @ REPL[109]:1
```

### PR DESCRIPTION
**Pull request by odow**
Closes #2655

### CODE REVIEW

File: src/constraints.jl [hunk 1]
```diff
@@ -134,8 +134,19 @@ end
 
 Get a constraint's name attribute.
 """
-function name(con_ref::ConstraintRef{<:AbstractModel,<:_MOICON})
-    return MOI.get(con_ref.model, MOI.ConstraintName(), con_ref)::String
+function name(con_ref::ConstraintRef{<:AbstractModel,C}) where {C<:_MOICON}
+    model = owner_model(con_ref)
+    if !MOI.supports(backend(model), MOI.ConstraintName(), C)
```
Comment (blegat) [inline src/constraints.jl:139]: Normally, set says if names can be `set`, not `get`. I guess for read-only model we can make an exception but it's another argument for extending supports (I can't find the issue where we discussed this)

Comment (odow) [reply src/constraints.jl:139]: If you can't `set`, I don't see how we can expect `get` to return something sensible here.

Comment (blegat) [reply src/constraints.jl:139]: What about `read_from_file` ?

Comment (odow) [reply src/constraints.jl:139]: That's not relevant at the JuMP level?

Read-only models should define `supports = true`, but throw a `SetAttributeNotAllowed` if `set` is called.

Comment (blegat) [reply src/constraints.jl:139]: Yes I agree, so we should clarify `supports`. We discussed about this but I can't find where. The closest is here: https://github.com/jump-dev/MathOptInterface.jl/issues/1310 but I don't think it's that.

**Review by blegat — APPROVED** [review]

File: src/constraints.jl (hunk 1 above)
Comment (odow) [reply src/constraints.jl:139]: https://github.com/jump-dev/MathOptInterface.jl/issues/1628

**Pull request merged**